developmentseed/lonboard
# Replace tooltip with side panel

This addresses #586 by adding a scrollbar to the tooltip when there is a long list of properties. Work in progress.

We discussed adding a side panel for browsing the properties, but I wanted to explore an alternative that enhances the existing tooltip rather than introducing a new UI element.

In this implementation, the `getTooltip` property in Deck.gl has been replaced with a custom tooltip rendered separately. This change was necessary because the trigger for displaying the tooltip was modified from mouse hover to mouse click. As a result, the tooltip persists when the mouse pointer moves away, allowing the user to scroll through the available properties.

### Known Issues:

- This is based off release `0.10.0-beta.2` commit, the main branch appears to be broken due to dependencies updates
- A horizontal scrollbar is always displayed alongside the vertical one due to an issue determining the table width, this needs an CSS update
- The tooltip remains in the same position when the map is panned. We could either hide it or have it move along with the map
- We are moving away from a Deck.gl-based solution. It might be worth implementing this as a custom widget, but the complexity of doing so is unclear at this point

### How to test

Open examples `north-america-roads` and `internet-speeds` and click on a rendered feature. The popup should be displayed and the table should be scrollable when the list is long.

cc @kylebarron @batpad
